specification.md

```diff
@@ -567,15 +602,24 @@ that the following state model is adhered to.
 Job instances are, in this API, stateful objects.  A job's state can be
 inspected via the `job.getStatus()` method which will return a `JobStatus`
 instance on which the job's state is available as an attribute.  State
-transitions can also be received via callbacks
+transitions can also be received via callbacks. The transitions through 
+`STAGE_IN`, `STAGE_OUT`, and `CLEANUP` must be generated even if a particular 
+job does not specify any files for staging or clean up.
```

Comment:
Yes, I think this is important as it makes the transitions more predictable and the application code handling state changes becomes much simpler.